tools/clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1232,7 +1232,7 @@ llvm::Constant *CodeGenModule::EmitConstantInit(const VarDecl &D,
   assert(E && "No initializer to emit");
 
   llvm::Constant* C = ConstExprEmitter(*this, CGF).Visit(const_cast<Expr*>(E));
-  if (C && C->getType()->isIntegerTy(1)) {
+  if (C && C->getType()->getScalarType()->isIntegerTy(1)) { // HLSL Change
```

Comment:
To handle bool vectors

Reply:
Shouldn't this also handle matrix here?

Reply:
Or will that insert an implicit cast that gets handled elsewhere?  If so, does that not happen for vector as well?  What scenario hits this case?

Reply:
This case happens when a static initializer is of type bool or bool vector: `static bool foo = { true };` or `static bool2 foo = { true, false };` Emitting the initializer will result in a register-representation i1 or <2 x i1>, which should be converted to its memory representation before storage to match the storage type of the variable and later support loads from it.